workshops/views.py

```diff
@@ -2680,3 +2733,31 @@ def duplicates(request):
     }
 
     return render(request, 'workshops/duplicates.html', context)
+
+
+#------------------------------------------------------------
+
+@login_required
+def certificate_download(request, certificate_id):
+    """Generate a certificate, and return the pdf for download.
+    Used on the person edit page."""
+
+    certificate = get_object_or_404(Certificate, pk=certificate_id)
+
+    filename = os.path.join(settings.CERTIFICATES_DIR,
+                            str(certificate.id) + '.pdf')
+    try:
+        binary = open(filename, 'rb')
+    except FileNotFoundError:
+        messages.error(
+            request,
+            'This certificate is not yet available for download. Please check back later.',
+        )
+        return render(request, 'workshops/certificate_error.html')
+
+    response = HttpResponse(binary, content_type='application/pdf')
+    response['Content-Disposition'] = 'inline; filename="{}.pdf"'\
+        .format(certificate.person.get_full_name())
```

Comment:
Filename should contain badge name too (can be short, ie. 'swc' / 'swc-instructor').